tools/ci.sh

```diff
@@ -29,6 +29,30 @@ function ci_code_formatting_run {
     tools/codeformat.py -v
 }
 
+########################################################################################
+# commit formatting
+
+function ci_commit_format_check {
+    # read from pipe
+    read sha
+    # test subject matches format we want
+    msg=$(git log -1 "$sha" --pretty=format:"%s")
+    echo $msg | grep -q -E '^.+: [A-Z]+.+ .+\.$'
```

Comment:
Is this better or more generally used?

Reply:
It avoids shipping stuff through grep, which has issues around flags on various platforms IIRC. I guess in general if it can be done cleanly in bash, why fork another process? (It's not a broadly known bash feature but I've been using it for many years.)

Reply:
Ok makes sense, the less different things needed the better. However see below: I might give up on using bash all together..